### PR TITLE
[SMALLFIX] Reset master state before recovering from  a checkpoint.

### PR DESCRIPTION
**Pull request by yuzhu**
We don't reset master state when backup masters read checkpoints. This gets us into trouble if a checkpoint is read after startup. For us to read a checkpoint after startup, the following needs to happen:

1. Master A starts
2. A reads the latest checkpoint.
3. An edit log is completed by the primary master
4. Another edit log is completed by the primary master
5. Before A reads the first new log, another backup master reads both logs and writes a checkpoint.
6. A sees the new checkpoint and reads the checkpoint instead of the edit logs

### CODE REVIEW

**Comment by yuzhu** [comment]
jenkins, test this please

**Comment by gpang** [comment]
@aaudiber can you take a look at this?

**Comment by gpang** [comment]
@yuzhu could you describe the issue with more details, in the description of this PR?

File: core/server/common/src/main/java/alluxio/master/journal/ufs/UfsJournal.java [hunk 1]
```diff
@@ -383,6 +383,9 @@ private long catchUp(JournalReader journalReader) {
       if (entry == null) {
         return journalReader.getNextSequenceNumber();
       }
+      if (journalReader.isReadingNewCheckpoint()) {
```
Comment (aaudiber) [inline core/server/common/src/main/java/alluxio/master/journal/ufs/UfsJournal.java]: Could we determine this by the sequence number of the read journal entry? If the sequence number is `0`, we call reset.

Comment (yuzhu) [reply core/server/common/src/main/java/alluxio/master/journal/ufs/UfsJournal.java]: done

**Comment by yuzhu** [comment]
@aaudiber looked into other cases where state is reset, seems like we make assumptions that whenever we process a root node, that we always reset. The test cases also rely on this fact. I think it is safer not to remove them at this point. `resetState` is idempotent anyways.

**Comment by aaudiber** [comment]
The failing tests are unrelated and being fixed by a separate PR

**Pull request merged**